src/extension.ts

```diff
@@ -111,14 +108,34 @@ async function attach(context: vscode.ExtensionContext, viaConfig: boolean) {
 
         responseArray.forEach(i => {
             i = utils.fixRemoteUrl(hostname, port, i);
-            items.push({ label: i.title, description: i.url, detail: i.webSocketDebuggerUrl });
+            items.push({ 
+                label: i.title, 
+                description: i.url, 
+                detail: i.webSocketDebuggerUrl 
+            });
         });
 
-        vscode.window.showQuickPick(items).then((selection) => {
-            if (selection) {
-                DevToolsPanel.createOrShow(context, selection.detail as string);
+        let targetMatch:boolean = false;
+        let targetWebsocketUrl:string = '';
+        if (typeof targetUrl === 'string' && targetUrl.length > 0 && targetUrl != defaultUrl) {
+            const matches = items.filter(i => targetUrl.localeCompare(i.description, 'en', { sensitivity: 'base' }) == 0);
+            if(matches && matches.length > 0 ){
+                targetWebsocketUrl = matches[0].detail;
+                targetMatch = true;
```

Comment:
You could probably just make do with targetWebsocketUrl and checking that is set to something, instead of the extra bool

Reply:
I added the extra bool so it was easier to read. That the intent of the code was to find a match and I felt like adding a bool made that clearer for future ppl. Happy to change it though.